### PR TITLE
feat(cmd/registry/auth/basic): allow to specify credentials as cli flags

### LINKED ISSUE #134: falcoctl registry login: enhancements
`falcoctl registry login` is used to login to a remote OCI registry. It accepts an argument which is the `domain name` of the registry and prompts the user to input its `username` and `password`:
```bash
$ falcoctl registry login my-regirstry.me
Username: my-username
Password: my-password
```
The credentials passed to the command are saved in `/home/user/.docker/config.json`. Subsequent calls to `falcoctl registry` commands will retrieve and use those credentials.

**What would you like to be added**:
The following flags/options should be added to the `login` command:
* `-p, --password`: registry password or identity token used for authentication;
* `-u, --username`: registry username used for authentication.

#### Command behavior
When no flags are passed to the `falcoctl registry login` command it must prompt the user for credentials. Otherwise, it will use the flag values and prompt the user for the missing ones.

**Why is this needed**:
This feature allows the `falcoctl registry login` to be scriptable.

### PR DESCRIPTION
**Pull request by maxgio92**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area cli

**What this PR does / why we need it**:

Introduces credentials CLI flags for the registry basic HTTP authentication.

**Which issue(s) this PR fixes**:

Fixes #134 

**Special notes for your reviewer**:

NA

### CODE REVIEW

**Comment by maxgio92** [comment]
Hey @alacuku I've just proposed this to accomplish #134. I assumed you referred to the basic authentication in that enhancement issue. If not, please let me know if I missed something.
Thank you

**Comment by alacuku** [comment]
Hi @maxgio92, that's correct the issue is about the `basic auth`.

The linter is complaining, could you have a look and let me know when it's ready for review?

**Comment by maxgio92** [comment]
Hey @alacuku, sorry for my late response, I missed it. The PR is now ready

File: cmd/registry/auth/basic/basic.go [hunk 1]
```diff
@@ -74,11 +81,22 @@ func (o *loginOptions) RunBasic(ctx context.Context, args []string) error {
...
+	if err := basic.Login(ctx, client, credentialStore, reg, o.username, o.password); err != nil {
 		return err
 	}
 	o.Printer.Verbosef("credentials added to credential store")
 	o.Printer.Success.Println("Login succeeded")
 
 	return nil
 }
+
+func (o *loginOptions) ensureCredentials() error {
```
Comment (alacuku) [inline cmd/registry/auth/basic/basic.go:93]: Could we support the use case where the username is given on the command line flag and the password from stdin?

**Comment by leogr** [comment]
Hey @maxgio92 

Any update on this?

I guess the PR needs to be rebased and also we should address this comment https://github.com/falcosecurity/falcoctl/pull/314#pullrequestreview-1684392862

**Comment by alacuku** [comment]
#588 supersedes this pr.

**Comment by alacuku** [comment]
hey @maxgio92, we can close this PR since thee #588 has been merged.
/close

**Comment by maxgio92** [comment]
Thank you @alacuku, my bad.

**Pull request closed without merging**